### ISSUE TITLE
Graph Structured Network for Image-Text Matching

### ISSUE THREAD

**Issue by hkefka385** [hkefka385/paper_reading#57]
## 0. 論文 
Journal/Conference: **CVPR 2020** 
Title: **Graph Structured Network for Image-Text Matching**
Authors: Chunxiao Liu, Zhendong Mao, Tianzhu Zhang, Hongtao Xie, Bin Wang, Yongdong Zhang
URL: https://openaccess.thecvf.com/content_CVPR_2020/html/Liu_Graph_Structured_Network_for_Image-Text_Matching_CVPR_2020_paper.html
## 1. どんなもの？ 
Image text matchingタスクを取り組むために，画像とテキストのマッチングを、グラフのマッチングとして扱った対応付け手法グラフ構造化マッチングネットワーク (GSMN) を提案
例︰
<img width="464" alt="スクリーンショット 2020-07-13 5 03 00" src="https://user-images.githubusercontent.com/10390578/87255497-40536f80-c4c6-11ea-8341-9d2b9fdd6c62.png">

## 2. 先行研究と比べてどこがすごい？ 
・テキストはdependency（構造解析），画像はobject間の距離をノードとして捉えグラフを自動的に構築
・グラフの情報を近接度などを用いて交換することで潜在特徴を更新
モデル図
<img width="910" alt="スクリーンショット 2020-07-13 5 03 05" src="https://user-images.githubusercontent.com/10390578/87255496-3a5d8e80-c4c6-11ea-839b-5b977fd43468.png">
## 4. どうやって有効だと検証した？ 
Flickr30KとMSCOCOの2つのベンチマークにおいて、提案手法がSOTAを更新
## メモ
従来の画像とテキストマッチング：画像中の物体と文中の単語間での粗い対応付けを行ってきた
→ グラフ構造化マッチングネットワーク (GSMN)を提案：オブジェクト・属性・関係を構造化フレーズとして明示的にモデル化 / ノードレベルと構造レベルのグラフをfine-grainedなマッチング

1 Introduction
Image text matchingのタスク
Fig1: グラフを用いてフレーズの対応関係を学習

テキストと画像の対応関係学習：損失関数の設定
トリプレット損失，adversarial loss，classification loss (Ref. 15，27，29)
トリプレット損失：Liwei Wang, Yin Li, Jing Huang, and Svetlana Lazebnik.Learning two-branch neural networks for image-text match-ing tasks.IEEE Transactions on Pattern Analysis and Ma-chine Intelligence, 41(2):394–407, 2019

領域とキーワードの対応関係を学習
・lightweight saliency model：Zhong Ji,  Haoran Wang,  Jungong Han,  and YanweiPang. Saliency-guided attention network for image-sentencematching.arXiv preprint arXiv:1904.09471, 2019
領域と単語を相関させ：Kuang-Huei Lee, Xi Chen, Gang Hua, Houdong Hu, and Xi-aodong He. Stacked cross attention for image-text matching.InProceedings of the European Conference on Computer Vi-sion (ECCV), pages 201–216, 2018
簡単な単語と領域を除去：Chunxiao Liu, Zhendong Mao, An-An Liu, Tianzhu Zhang,Bin Wang, and Yongdong Zhang. Focus your attention: Abidirectional focal attention network for image-text match-ing. InProceedings of the 27th ACM International Confer-ence on Multimedia, pages 3–11. ACM, 2019

オブジェクト構造を学習するための限界
・関係と属性の対応関係を学習が難しい
・記述と属性の対応関係を学習の必要

提案モデル：細分化された対応関係を推定するGraph Structured Matching Network (GSMN)
画像とテキストをそれぞれのグラフを構築し，ノードと構造レベルのマッチングを行う
画像とテキストのペアの類似度を予測

貢献
・細かいフレーズ応答を学習してマッチングを行うモデルを提案 / 画像とテキストの異なるグラフ同士でマッチングを行うフレームワーク
・GCNを用いてノードの対応を伝搬
・Flickr30K，MSCOCOを用いた実験

2 Related work
Text-image matchingのタスク︰global correspondenceとlocal correspondenceの2種類に分割可能

・global correspondence learning (Ref.19,4,29,29,31,22,34,1)：一致した画像と文のペアの類似度を最大化
・Ref19：残差ブロックの利用
・画像やテキスト間のグラフ構造を保存し，対応を学習
Yiling Wu, Shuhui Wang, and Qingming Huang. Learningsemantic structure-preserved embeddings for cross-modalretrieval. In2018 ACM Multimedia Conference, pages 825–833. ACM, 2018

・local correspondence learning
Yan Huang, Wei Wang, and Liang Wang. Instance-aware im-age and sentence matching with selective multimodal lstm.InProceedings of the IEEE Conference on Computer Visionand Pattern Recognition, pages 2310–2318, 2017.
Yan Huang, Qi Wu, Chunfeng Song, and Liang Wang.Learning semantic concepts and order for image and sen-tence matching.  InProceedings of the IEEE Conferenceon Computer Vision and Pattern Recognition, pages 6163–6171, 2018.
Zhong Ji,  Haoran Wang,  Jungong Han,  and YanweiPang. Saliency-guided attention network for image-sentencematching.arXiv preprint arXiv:1904.09471, 2019.

共通の空間に写像︰Lin Ma, Zhengdong Lu, Lifeng Shang, and Hang Li. Mul-timodal convolutional neural networks for matching imageand sentence. InICCV, pages 2623–2631, 2015
Attentionを用いたSCANモデル：Kuang-Huei Lee, Xi Chen, Gang Hua, Houdong Hu, and Xi-aodong He. Stacked cross attention for image-text matching.InProceedings of the European Conference on Computer Vi-sion (ECCV), pages 201–216, 2018
→ 粗い対応関係を設定

本研究では、より細かいフレーズの対応関係を学習：ノードが相互作用しているかどうかを識別しシーングラフを生成

3 Method
Visionとテキストの特徴からグラフを構築 → ノードの対応関係や構造レベルでのマッチングを行う

3.1 Graph Construction
・Textual Graph
Stanford CoreNLPでdependencyを取得
依存があればエッジが存在
単語の類似度行列 (e.q.1) (softmax)をエッジの強さとして定義

・Visual Graph
Faster-RCNNで検出された領域をノードとして，すべての領域に対してエッジで結びつける
Will Norcliffe-Brown, Stathis Vafeias, and Sarah Parisot.Learning conditioned graph structures for interpretable vi-sual question answering. InAdvances in Neural InformationProcessing Systems, pages 8334–8343, 2018
と同様に，polar coordinate を用いて各画像の空間関係をモデル化
pair-wise polar coordinatesを用いてノード間のエッジの重みを設定

3.2 Multimodal Graph Matching
画像とテキストのglobal similaritを算出するためのノードレベル + structure-levelのマッチング

3.2.1 Node-level matching
Visual nodeとtext nodeの類似度を計算し，それをvisual nodeの重みとしてsoftmax関数に適用 (e.q,3)
→ ブロックごとの類似度を計算するモジュール
I番目ｎtext nodeの特徴と，それに対応したvisual nodeをtblockに分割し類似度を測定？ →Matchin vectorの生成 (e.q.4)

3.2.2 Structure level matching
GCN層で各ノードのマッチングベクトルを近傍のマッチングベクトルを用いて更新 (e.q.6)
Structureどうしの一致の定式化(e.q.7-8)

4
4.1
 Flickr30K and MSCOCOデータセットを用いて実験
